overlord/devicestate/devicestate_test.go

```diff
@@ -3184,6 +3184,161 @@ func (s *deviceMgrSuite) TestDoRequestSerialReregistrationStreamFromService(c *C
 	c.Assert(err, IsNil)
 }
 
+func (s *deviceMgrSuite) TestDoRequestSerialReregistrationIncompleteStreamFromService(c *C) {
+	mockServer := s.mockServer(c, "REQID-1", nil)
+	defer mockServer.Close()
+
+	restore := devicestate.MockBaseStoreURL(mockServer.URL)
+	defer restore()
+
+	// will produce an incomplete stream!
+	s.ancillary = s.brands.AccountsAndKeys("rereg-brand")[:1]
+
+	// setup state as done by first-boot/Ensure/doGenerateDeviceKey
```

Comment:
could you also add a comment where this setting up finishes?

Reply:
actually these very copy pasted and don't really reflect what is happening here, what we setup here is the full state after the initial registration. I changed these comments